proposals/Log-Based-Registry.md

```diff
@@ -0,0 +1,211 @@
+# Log Based Registry
+
+# Goals
+
+* Reduce required trust in registry operator
+* Make malicious activity detectable
+* Enable package state replication (mirroring)
+
+# Package Log
+
+For each package, a registry maintains an append-only package log.
+This log is the canonical record of the package state and contains cryptographic information that makes it possible to verify its integrity.
+Information can be derived from the log and used by some consumers, but other consumers and auditors will want to directly read and verify the log itself.
+
+## Package Entry
+
+A package log is a sequence of typed Package Entries which represent events in the life of a package.
+
+```protobuf
+message PackageEntry {
+    // The previous entry in the log.
+    // First entry of a log has no previous entry.
+    optional string prev = 1;
+    // The warg protocol version.
+    // Encoded using semver 2.0.
+    // Only legal value currently is "0.1.0".
+    string version = 2;

```

Comment:
Semver is awkward for protocol versioning. I would stick to an unsigned int here.